app/models/patient.rb

```diff
@@ -740,23 +740,54 @@ def self.order_by_risk(asc: true)
   end
 
   # Check for potential duplicate records. Duplicate criteria is as follows:
-  # - matching first name, last name, sex, and DoB
+  # - matching values of first name and last name. Optionally sex and DoB can also match for more detailed messaging
+  # e.g. Jon Smith M null would be a duplicate of Jon Smith M 1/1/2000
   # OR
   # - matching state/local id
   def self.duplicate_data(first_name, last_name, sex, date_of_birth, user_defined_id_statelocal)
-    dup_info = where('first_name = ?', first_name)
-               .where('last_name = ?', last_name)
-               .where('sex = ?', sex)
-               .where('date_of_birth = ?', date_of_birth)
+    # Track which matches have occurred
+    duplicate_field_data = []
 
+    # check for a duplicate state/local id
     dup_statelocal_id = where('user_defined_id_statelocal = ?', user_defined_id_statelocal&.to_s&.strip)
```

Comment:
Slight issue here. We're gonna have to also remove leading/trailing whitespace on the values we're checking against in the database, since they're not stripped on the way in in import. For example, we could have values like `" EX-123   "` in the database, which this wouldn't match to `"EX-123"`.

```suggestion
    dup_statelocal_id = where('trim(user_defined_id_statelocal) = ?', user_defined_id_statelocal&.to_s&.strip)
```

That should do the trick, although I'm not sure about the performance implication there... Is this a bad idea? Would we need to create an index it? @Bialogs @holmesie @timwongj 

Also please add a unit test case for this scenario.

Reply:
Added this change